### PR TITLE
Digix

### PR DESCRIPTION
**Pull request by talbaneth**

### CODE REVIEW

File: contracts/mockContracts/MockDGXDex.sol [hunk 1]
```diff
@@ -43,91 +33,102 @@ contract MockDGXDEX is Withdrawable {
...
         require(token.transferFrom(msg.sender, this, amount));
         msg.sender.transfer(amountWei);
         
         success = true;
 
-        Sell(success, amountWei);
+        Sell(success, amountWei, msg.sender, weiPerDgxMg);
     }
 
     function verify_signed_price(uint _block_number, uint _nonce, uint _price, address _signer, bytes _signature)
```
Comment (ilanDoron) [inline contracts/mockContracts/MockDGXDex.sol]: wrong notation.

File: test/mockDGXDEX.js [hunk 2]
```diff
@@ -1,11 +1,11 @@
-var TestToken = artifacts.require("./mockContracts/TestToken.sol");
-var mockDGXDEX = artifacts.require("./mockContracts/MockDGXDEX.sol")
-var Helper = require("./helper.js")
+let TestToken = artifacts.require("./mockContracts/TestToken.sol");
+let mockDGXDEX = artifacts.require("./mockContracts/MockDGXDEX.sol")
+let Helper = require("./helper.js")
 
-var myToken;
+let myToken;
```
Comment (ilanDoron) [inline test/mockDGXDEX.js]: is this used?

File: test/mockDGXDEX.js [hunk 3]
```diff
@@ -1,11 +1,11 @@
-var TestToken = artifacts.require("./mockContracts/TestToken.sol");
-var mockDGXDEX = artifacts.require("./mockContracts/MockDGXDEX.sol")
-var Helper = require("./helper.js")
+let TestToken = artifacts.require("./mockContracts/TestToken.sol");
```
Comment (ilanDoron) [inline test/mockDGXDEX.js]: I think here const is better.
all these values will not change.

File: test/mockDGXDEX.js [hunk 4]
```diff
@@ -1,11 +1,11 @@
...
+let mockDGXDEX = artifacts.require("./mockContracts/MockDGXDEX.sol")
+let Helper = require("./helper.js")
 
-var myToken;
+let myToken;
 
-var ethAddress = '0x00eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee';
-const BigNumber = require('bignumber.js');
+let ethAddress = '0x00eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee';
```
Comment (ilanDoron) [inline test/mockDGXDEX.js]: const

File: test/mockDGXDEX.js [hunk 5]
```diff
@@ -1,11 +1,11 @@
...
+let BigNumber = require('bignumber.js');
```
Comment (ilanDoron) [inline test/mockDGXDEX.js]: const

File: contracts/mockContracts/MockDGXDex.sol [hunk 6]
```diff
@@ -43,91 +33,102 @@ contract MockDGXDEX is Withdrawable {
...
-        if(blockCheck) {
-            require((block_number + 5) >= block.number);
+        if (blockCheck) {
+            require((blockNumber + BLOCK_EXPIRATION) >= block.number);
         }
 
-        require((block_number + 5) >= block.number);
+        amountMg = (msg.value / weiPerDgxMg);
+        amountMgDgxWei = amountMg * DGX_DECIMALS;
+        amountDgxWei = amountMgDgxWei / GRAM_TO_MILLIGRAM;
```
Comment (ilanDoron) [inline contracts/mockContracts/MockDGXDex.sol]: so. let me understand.
wei represents Gram and not MG?

File: contracts/mockContracts/MockDGXDex.sol [hunk 7]
```diff
@@ -43,91 +33,102 @@ contract MockDGXDEX is Withdrawable {
...
-            
-            (_verifies,_actual_signer) = ECVerify.ecrecovery(_hash, _signature);
-            _verified = (_verifies && (_actual_signer == _signer));
-        }
+    public
+    returns (bool _verified, address _actual_signer)
+    {
+        bytes32 _hash;
+        bool _verifies;
+        _hash = hash_price_data(_block_number, _nonce, _price);
```
Comment (ilanDoron) [inline contracts/mockContracts/MockDGXDex.sol]: hash_price_data wrong notation.

By the way. indentation and notation are clearly explained in solidity documents.
also u can run a tool - solhint that issues errors regarding indentation...
https://github.com/protofire/solhint

File: contracts/mockContracts/MockDGXDex.sol (hunk 1 above)
Comment (talbaneth) [reply contracts/mockContracts/MockDGXDex.sol]: this is code that we copied from Digix contract.
So I don't want to fix it.
It's good to have it distingushed in our code and I added the comment:
 /* The code below is DGX signature verification (logic and code style). */

File: test/mockDGXDEX.js (hunk 2 above)
Comment (talbaneth) [reply test/mockDGXDEX.js]: removed.

File: test/mockDGXDEX.js (hunk 3 above)
Comment (talbaneth) [reply test/mockDGXDEX.js]: ok, but note that searching "const" smart contracts in test dir found only one occurence (and lots of "var" :)). So you don't really use it..

File: test/mockDGXDEX.js (hunk 5 above)
Comment (talbaneth) [reply test/mockDGXDEX.js]: ditto.

File: test/mockDGXDEX.js (hunk 4 above)
Comment (talbaneth) [reply test/mockDGXDEX.js]: ditto.

File: contracts/mockContracts/MockDGXDex.sol (hunk 7 above)
Comment (talbaneth) [reply contracts/mockContracts/MockDGXDex.sol]: ditto. It's digix contract code.

**Pull request merged**